ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/TransactionTransmitter.java

```diff
@@ -95,66 +107,64 @@ private JsonRpcBody createSignedTransactionBody() {
   }
 
   private void sendTransaction(final Buffer bodyContent) {
-    final HttpServerRequest httpServerRequest = context.getInitialRequest();
+    final HttpServerRequest httpServerRequest = sendTransactionContext.getInitialRequest();
     final HttpClientRequest request =
         ethNodeClient.request(
-            httpServerRequest.method(), httpServerRequest.uri(), this::handleResponse);
+            httpServerRequest.method(),
+            httpServerRequest.uri(),
+            response -> handleResponse(sendTransactionContext.getRoutingContext(), response));
 
-    request.headers().setAll(httpServerRequest.headers());
-    request.headers().remove("Content-Length"); // created during 'end'.
-    request.setChunked(false);
-    request.end(bodyContent);
+    sendRequest(request, bodyContent, sendTransactionContext.getRoutingContext());
   }
 
-  private void handleResponse(final HttpClientResponse response) {
-    logResponse(response);
-    response.bodyHandler(
-        body -> {
-          logResponseBody(body);
-          handleResponseBody(response, body);
-        });
+  private void exceptionHandler(final RoutingContext context, final Throwable thrown) {
+    LOG.info("An exception was thrown by the transaction submission, {}", thrown);
+    if (thrown instanceof TimeoutException) {
```

Comment:
These two cases could be put into a single `if` clause: `if( thrown instanced TimeoutException || thrown instanced ConnectException)`

Reply:
+1 agreed - but this should be removed as is used from base class.